docs/write-code/PERFORMANCE.md

```diff
@@ -171,3 +171,39 @@ const printAge(_name: string, age: number){
     console.log(age);
 }
 ```
+
+## Utilize React Hooks
+
+React hooks are a great way to optimize your components. Always try to use hooks like `useMemo`, `useCallback`, `useEffect` etc. to optimize your components.
+
+### Use `useRequiredScopes` hook instead of `hasRequiredScopes` function
+
+The `useRequiredScopes` hook is a more optimized way to check for required scopes in the application. Always use this custom hook instead of the `hasRequiredScopes` function.
+
+:white_check_mark: Do
+
+```typescript
+import { useRequiredScopes } from "@wso2is/access-control";
+
+const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+
+const hasUsersUpdatePermissions: boolean = useRequiredScopes(
```

Comment:
Often we need to check these permissions outside react components (eg: in util functions etc.), but we cannot use hooks in those places. Is there a non-hook equivalent to use outside react functional components?

Reply:
When using outside of the react components, we can use`hasRequiredScopes` function instead. But ideally we should try to avoid using this function altogether. I have added some docs with b938d606313062ee0d0770c25215665021fe0df4